src/components/Publication/Publication.js

```diff
@@ -76,31 +76,28 @@ class Publication extends Component {
     };
 
     this.setSubdoc = this.setSubdoc.bind(this);
+    this.setArethusaLoaded = this.setArethusaLoaded.bind(this);
 
     this.arethusa = new ArethusaWrapper();
   }
 
   componentDidMount() {
-    // eslint-disable-next-line no-undef
-    this.interval = window.setInterval(this.setSubdoc, 100);
+    document.body.addEventListener('ArethusaLoaded',this.setArethusaLoaded);
   }
 
   componentWillUnmount() {
-    // eslint-disable-next-line no-undef
-    window.clearInterval(this.interval);
+    document.body.removeEventListener('ArethusaLoaded');
   }
 
+  setArethusaLoaded() {
+    document.body.removeEventListener('ArethusaLoaded');
```

Comment:
I'm getting the error `Failed to execute 'removeEventListener' on 'EventTarget': 2 arguments required, but only 1 present` in my console from this line.